bigchaindb_driver/transport.py

```diff
@@ -50,11 +74,28 @@ def forward_request(self, method, path=None,
 
         """
         connection = self.get_connection()
-        response = connection.request(
-            method=method,
-            path=path,
-            params=params,
-            json=json,
-            headers=headers,
-        )
-        return response.data
+        if connection is not None:
+            try:
+                response = connection.request(
+                    method=method,
+                    path=path,
+                    params=params,
+                    json=json,
+                    headers=headers,
+                )
+                self.pool.success_node()
+                return response.data
+            except BaseException as err:
+                self.pool.fail_node()
+                return self.forward_request(
+                    method=method,
+                    path=path,
+                    params=params,
+                    json=json,
+                    headers=headers,
+                    error=err)
+        elif error is not None:
+            raise error
```

Comment:
What do you think is better? Raising `err` (that is _the last error_) or `error` (that is _the first error_)?

Reply:
We could not raise```err``` as at that point is not available locally and it will throw
```
UnboundLocalError: local variable 'err' referenced before assignment
```